runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8124,7 +8124,7 @@ done:;
 
 public:
 
-#if defined(J9VM_ARCH_X86) && defined(__GNUC__) && ((__GNUC__> 4) || (__GNUC__ == 4 && __GNUC_MINOR__ > 6))
+#if (defined(J9VM_ARCH_X86) || defined(S390)) && defined(__GNUC__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ > 6)))
```

Comment:
At this point do we want to just remove the VMINLINE all of the time? With these changes it will be removed from Linux x86 and Linux 390 when using any new GCC. On OS X with clang and computed goto enabled (the new default) the run method is NOT inlined even with VMINLINE specified.

My assumptions are:
1. now that we compile the BytecodeInterpreter with clang on Windows if computed goto is enabled it will increase performance and force the run method to not be inlined.
2. If we try computed goto on Linux P with GCC 7 computed goto will improve performance as well so we will need to add that to this list of ever growing defines that turn off the VMINLINE
3. Removing the VMINLINE all of the time likely only impacts platforms where the BytecodeInterpreter is compiled with xlC. Are there any of those platforms anymore?

Reply:
AIX and z/OS are the only platforms that compile the BytecodeInterpreter with xlC.
Note Windows isn't using clang yet, not until #2979 is merged.